scripts/check_changelog.js

```diff
@@ -0,0 +1,88 @@
+const { execSync } = require('child_process');
+const path = require('path');
+const fs = require('fs');
+
+// Colors for output
+const colors = {
+	red: text => `\x1b[31m${text}\x1b[0m`,
+	green: text => `\x1b[32m${text}\x1b[0m`,
+};
+
+try {
+	// Get all staged files
+	const stagedFiles = execSync('git diff --cached --name-only', { encoding: 'utf-8' })
+		.trim()
+		.split('\n')
+		.filter(Boolean);
+
+	// Initialize map to track packages with code changes and changelog updates
+	const packagesWithChanges = new Map();
+
+	// Scan all staged files
+	stagedFiles.forEach(file => {
+		// Get package name
+		const packageMatch = file.match(/packages\/([^/]+)/);
+		if (!packageMatch) return;
+
+		const packageName = packageMatch[1];
+
+		// Check if it's a code file
+		const isCodeFile = /\.(js|jsx|ts|tsx|css|json)$/.test(file);
```

Comment:
```suggestion
		const isCodeFile = /\.(js|jsx|ts|tsx|css|json)$/.test(file);
```
It should look for changes in only `ts` and `json` files, and it should skip looking any changes in test dirs